cloudinit/sources/DataSourceVMware.py

```diff
@@ -146,7 +145,8 @@ def __init__(self, sys_cfg, distro, paths, ud_proc=None):
 
         self.cfg = {}
         self.data_access_method = None
-        self.vmware_rpctool = VMWARE_RPCTOOL
+        self.rpctool = None
```

Comment:
I think this and the line after still have some potential to result in tracebacks. If neither is found or [we have envvars](https://github.com/canonical/cloud-init/blob/fc7b92ba7529cf749722bd5345c9bfdd6d4ca5fd/cloudinit/sources/DataSourceVMware.py#L159) (such that they never get a chance to get set), these will stay None. Then when they're called from places like [here](https://github.com/canonical/cloud-init/blob/fc7b92ba7529cf749722bd5345c9bfdd6d4ca5fd/cloudinit/sources/DataSourceVMware.py#L216) or [here](https://github.com/canonical/cloud-init/blob/fc7b92ba7529cf749722bd5345c9bfdd6d4ca5fd/cloudinit/sources/DataSourceVMware.py#L240), we'll try to call `self.rpctool_fn` with arguments, but it will still be `None`, resulting in a TypeError.

Reply:
The first example cannot do that as the detected method is checked first in the redact function. I also just pushed an update that checks if the rpctool and rpctool_fn vars are set before using them in the advertise method.

Reply:
To be clear, on the advertise func, I want that to work even if the guestinfo platform wasn't chosen but the capability _exists_. That's why the check I added only validates the rpctool vars.

Reply:
And by the way, the entire purpose of the `self.rpctool_fn` was so that the function to invoke was decided exactly once and then the rest of the datasource never had to have another if statement about the logic.

Reply:
@TheRealFalcon I was waiting for you to respond before resolving this.

Reply:
@akutz , yeah, that works. Thanks